service/state/core.go

```diff
@@ -68,50 +67,24 @@ func NewCore(dir string) (*Core, error) {
 
 	var height uint64
 	var commit flow.StateCommitment
-	err = db.View(func(tx *badger.Txn) error {
-
-		// first we get the last commit
-		if err := storage.RetrieveLastCommit(&commit)(tx); err != nil {
-			return fmt.Errorf("could not retrieve last commit: %w", err)
-		}
-
-		// then we get the height associated with it
-		if err := storage.RetrieveHeightByCommit(commit, &height)(tx); err != nil {
-			return fmt.Errorf("could not retrieve height from commit: %w", err)
-		}
-		return nil
+	trie, err := trie.NewEmptyMTrie(pathfinder.PathByteSize)
+	if err != nil {
+		return nil, fmt.Errorf("could not initialize empty trie: %w", err)
+	}
+
+	empty := trie.RootHash()
```

Comment:
You can use `commit := trie.RootHash()` here. Then you don't need the check later imo. The second operation of the first combine should never fail if the first one succeeds.